subsys/bluetooth/host/hci_core.c

```diff
@@ -364,31 +365,41 @@ int bt_hci_cmd_send_sync(u16_t opcode, struct net_buf *buf,
 	net_buf_put(&bt_dev.cmd_tx_queue, buf);
 
 	err = k_sem_take(&sync_sem, HCI_CMD_TIMEOUT);
-	__ASSERT(err == 0, "k_sem_take failed with err %d", err);
+	if (err) {
+		BT_WARN("Failed to send command to driver (%d)", err);
+		/* These pointers are to the calling threads' stack variables
+		 * so clear them so that command complete handling will not
+		 * corrupt the stack. The command complete handling will check
+		 * for valid pointers.
+		 */
+		cmd(buf)->sync = NULL;
+		cmd(buf)->state = NULL;
+		net_buf_unref(buf);
+		return err;
+	}
 
-	BT_DBG("opcode 0x%04x status 0x%02x", opcode, cmd(buf)->status);
+	status = cmd(buf)->status;
+	if (status) {
+		BT_INFO("opcode 0x%04x status 0x%02x", opcode, status);
```

Comment:
I think BT_WARN() would be more appropriate

Reply:
Updated to BT_WARN